mesonbuild/dependencies/qt.py

```diff
@@ -252,11 +252,11 @@ class QmakeQtDependency(_QtBase, ConfigToolDependency, metaclass=abc.ABCMeta):
 
     """Find Qt using Qmake as a config-tool."""
 
-    tool_name = 'qmake'
     version_arg = '-v'
 
     def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any]):
         _QtBase.__init__(self, name, kwargs)
+        self.tool_name = f'qmake{self.qtver}'
         self.tools = [f'qmake{self.qtver}', f'qmake-{self.name}', 'qmake']
```

Comment:
Why are you overwriting tool_name with self.tools[0]? Your change below already is intended to make machine file lookups prioritize entries for self.tools before self.tool_name (because we invoke `find_external_program(name=self.tool_name, default_names=self.tools)`)

Reply:
The tool_name should be suffixed with the version at all times, IMO. It's clearer to the user since it's used as the display name. `qmake` is shipped with both Qt5 and Qt6 SDKs. `qmake5` `qmake6` are only shipped with the corresponding SDKs.

Reply:
I was mostly looking to better understand the change, I think your reasoning is fair.

Checking twice doesn't really matter IMO... it's just looking up machine file info and not stat'ing the filesystem. However it does no real harm (because you used an orderedset to preserve order).